packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -774,6 +789,38 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		}
 	}
 
+	/**
+	 * Delete unused storage blobs, if any. After GC runs and notifies blob manager of local blob ids that are sweep
+	 * ready, blob manager deletes them from the redirect table. This function is then called to delete the storage
+	 * ids of these blobs if they are not used anymore. A blob is unused if there is no entry from the local id to
+	 * storage id in the redirect table, i.e., all local id entries have been deleted.
+	 * @param maybeUnusedBlobIds - List of storage ids for blobs whose local id entries were just deleted and are
+	 * now candidates for deletion.
+	 */
+	private deleteUnusedStorageBlobs(maybeUnusedBlobIds: string[]) {
+		if (maybeUnusedBlobIds.length === 0) {
+			return;
+		}
+
+		const storageBlobsToDelete: Set<string> = new Set(maybeUnusedBlobIds);
+		for (const [localId, storageId] of this.redirectTable.entries()) {
+			// For a storage id that is a candidate for deletion, ignore storageId -> storageId entries. But if the
+			// redirect table contains an entry from a local id to it, it is still in use and should not be deleted.
+			if (
+				storageId !== undefined &&
+				storageBlobsToDelete.has(storageId) &&
+				localId !== storageId
+			) {
+				storageBlobsToDelete.delete(storageId);
```

Comment:
This code is clear what it's doing, but at first glance `storageBlobsToDelete.delete` sounds like it's doing the delete, not removing something not to delete.

Consider flipping this loop to add the ones that should be deleted rather than removing the ones that shouldn't be.  (I had this feedback in mind before noticing this amusing situation)

Reply:
At the very least update the comment that says "ignore storageId -> storageId entries" - it's a double negative? As in "Ignore when considering what not to delete", right?

Reply:
Ok that "ignore" comment is even more distracting now that I realize that the entire point of this function is to clean up those storageId -> storageId mappings (right?)

Reply:
Updated the code and comments. Take a look.